docs/extension-development/check-if-woo-is-active.md

```diff
@@ -4,28 +4,45 @@ menu_title: Check if WooCommerce is active
 tags: how-to
 ---
 
-You can wrap your plugin in a check to see if WooCommerce is installed:
+When developing for WooCommerce, it's important to check that WooCommerce is installed and active before your own code runs. This ensures no errors occur from missing WooCommerce functions or classes.
+
+There are a few methods to achieve this. The most reliable approach is to initiate your code on `plugins_loaded`. This means you can be sure that WooCommerce has already initiated, by checking if one of its core classes exists.
 
 ```php
-// Test to see if WooCommerce is active (including network activated).
+add_action( 'plugins_loaded', 'prefix_plugins_loaded' );
 
-$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . 'woocommerce/woocommerce.php';
+function prefix_plugins_loaded() {
+	// Check if the "WooCommerce" class exists. If not, exit early.
+	if ( ! class_exists( 'WooCommerce' ) ) {
+		return;
+	}
 
-if (
+	// Custom code here. WooCommerce is active and initiated...
+}
+```
 
-in_array( $plugin_path, wp_get_active_and_valid_plugins() )
+Another route you could take is to initiate your code on the `woocommerce_init` hook. We know this only runs when WooCommerce is active and initiated.
```

Comment:
```suggestion
Another route you could take is to initialize your code on the `woocommerce_init` hook. We know this only runs when WooCommerce is active and initialized.
```

Reply:
Optional thing—if this makes the doc too complex or is veering off track, feel free to ignore—but it could be worth adding a section that talks about version comparison. That is, highlighting this sort of approach:

```php
add_action( 'woocommerce_init', function () {
	// Only continue if we have access to version 8.7.0 or higher.
	if ( version_compare( wc()->version, '8.7.0', '<' ) ) {
		return;
	}

	// OK, we're good to go... (add rest of code here)
} );
```

This could be useful for folks who depend on functionality added after version x.y.z, or just as a subtle way to encourage version-awareness.

Reply:
I think we should include this - it's useful information that isn't included anywhere else